helper/schema/grpc_provider.go

```diff
@@ -281,6 +282,32 @@ func (s *GRPCProviderServer) ValidateResourceTypeConfig(ctx context.Context, req
 		resp.Diagnostics = convert.AppendProtoDiag(ctx, resp.Diagnostics, err)
 		return resp, nil
 	}
+	if req.ClientCapabilities == nil || !req.ClientCapabilities.WriteOnlyAttributesAllowed {
```

Comment:
We might need to automatically set the client capabilities in `terraform-plugin-go`, so that we don't have to do this null check.